src/main/java/org/amahi/anywhere/activity/ServerFileAudioActivity.java

```diff
@@ -161,6 +163,12 @@ private void prepareFiles() {
             serverFiles.add(serverFile);
             getIntent().putExtra(Intents.Extras.SERVER_FILES, new ArrayList<Parcelable>(serverFiles));
         }
+
+        NetworkUtils networkUtils = new NetworkUtils(this);
+        if (!networkUtils.isNetworkAvailable()) {
+            Toast.makeText(this, R.string.message_connect_and_try_again, Toast.LENGTH_SHORT).show();
+            finish();
+        }
```

Comment:
I think it will be better if you check the connection before opening the activity.

Reply:
Yeah I agree. Will change it.